### PR TITLE
Speed up livesync on android device

### PR DESCRIPTION
**Pull request by Fatme**
Related:  https://github.com/NativeScript/nativescript-cli/issues/921

### CODE REVIEW

File: lib/providers/device-app-data-provider.ts [hunk 1]
```diff
@@ -37,18 +39,34 @@ export class AndroidAppIdentifier extends deviceAppDataBaseLib.DeviceAppDataBase
...
 
 	public isLiveSyncSupported(): IFuture<boolean> {
 		return Future.fromResult(true);
 	}
+
+	private getSyncFolderName(): IFuture<string> {
+		return ((): string =>{
+			let adb =  this.$injector.resolve(AndroidDebugBridge, { identifier: this.device.deviceInfo.identifier });
+			let deviceHashService = this.$injector.resolve(AndroidDeviceHashService, {adb: adb, appIdentifier: this.appIdentifier});
+			let hashFile = this.$options.force ? null : deviceHashService.doesShasumFileExistsOnDevice().wait(); // introduce new method in deviceHashService and
```
Comment (rosen-vladimirov) [inline lib/providers/device-app-data-provider.ts]: remove this comment

File: lib/providers/device-app-data-provider.ts [hunk 2]
```diff
@@ -37,18 +39,34 @@ export class AndroidAppIdentifier extends deviceAppDataBaseLib.DeviceAppDataBase
...
+			return this.$options.watch || hashFile ? "sync" : "fullsync";
```
Comment (teobugslayer) [inline lib/providers/device-app-data-provider.ts]: magic constants. consider unmagic them.

**Comment by rosen-vladimirov** [comment]
ping @teobugslayer - I've changed hash file to be JSON (check the last commit)

**Comment by dtopuzov** [comment]
:+1:

**Comment by teobugslayer** [comment]
Squash and :+1:

**Pull request merged**